app/src/common/shared/com/igalia/wolvic/ui/viewmodel/WindowViewModel.java

```diff
@@ -53,7 +54,9 @@ public class WindowViewModel extends AndroidViewModel {
     private MutableLiveData<ObservableBoolean> isActiveWindow;
     private MediatorLiveData<ObservableBoolean> isTitleBarVisible;
     private MutableLiveData<Windows.ContentType> currentContentType;
+    public MutableLiveData<Windows.ContentType> lastContentType;
     private MediatorLiveData<ObservableBoolean> isNativeContentVisible;
+    private MutableLiveData<ObservableBoolean> backToNewTabEnabled;
```

Comment:
I see that you had to add new attributes in the `WindowViewModel` to keep track of the current state, specially when the user browses back and forward, shows and hides the library, etc. Perhaps not all of these are needed and we can use the current URL to select the appropriate UI and content type (but maybe that is just wishful thinking on my part :-) ).

One question that comes to mind is whether these need to be part of the `WindowViewModel` instead of being internal properties of `WindowWidget`, for example. The view model is specially useful to listen to changes in particular values and to update XML layouts automatically.

Another observation is that this code tries to manipulate how the Back and Forward navigation works, and the result can be a bit confusing sometimes. This happens specially when we combine back/forward with the Home button.

In this case, I think it is better to leave that behaviour as something internal to the Web engine (Gecko or Chromium). Personally, I often use desktop Firefox as the reference when trying to understand how this navigation should work,